.github/workflows/lint.yml

```diff
@@ -0,0 +1,44 @@
+---
+name: Lint
+
+on:
+  push:
+    branches: ["master"]
+
+  pull_request:
+    branches: ["master"]
+
+jobs:
+  tests:
+    name: "Python ${{ matrix.python-version }}"
+    runs-on: "ubuntu-latest"
+
+    strategy:
+      matrix:
+        python-version: ["3.11", "3.12"] # 3.13 doesn't seem to be supported by Pydantic (PyO3 error)
+
+    steps:
+      - uses: "actions/checkout@v4"
+
+      # Ref: https://docs.astral.sh/uv/guides/integration/github/#installation
+      - name: Setup uv
+        uses: astral-sh/setup-uv@v5
+        with:
+          enable-cache: true
+          cache-dependency-glob: "uv.lock"
+          python-version: ${{ matrix.python-version }}
+
+      - name: Install Dependencies
+        run: uv sync --all-extras --dev
+
+      - name: Check Copyrights
+        uses: apache/skywalking-eyes/dependency@main
+        with:
+          command: "check"
+          config: ".licenserc.yaml"
+
+      - name: "Linting"
+        run: "make lint-check"
+
+      - name: "Build Package"
+        run: "make build"
```

Comment:
I'm not seeing a task for testing with a matrix of python builds. Even if we only have a single unit test, it is valuable for us to test against all python versions such that at least we're syntactically correct for all supported Python versions

Reply:
I did not want to add the test task because there was no tests 😌 

I have added it in https://github.com/datarobot-oss/pulumi-datarobot-utils/pull/1/commits/22e6b529a74f7d821988b5930d456104cea29715 along with a couple of unit tests to make pytests run green.